pkg/prometheus/promcfg.go

```diff
@@ -398,28 +398,36 @@ func (cg *configGenerator) generateServiceMonitorConfig(version semver.Version,
 
 	// Relabel namespace and pod and service labels into proper labels.
 	relabelings = append(relabelings, []yaml.MapSlice{
-		{
-			{Key: "source_labels", Value: []string{"__meta_kubernetes_namespace"}},
-			{Key: "target_label", Value: "namespace"},
-		},
-		{
+		{ // Relabel node labels for pre v2.3 meta labels
 			{Key: "source_labels", Value: []string{"__meta_kubernetes_endpoint_address_target_kind", "__meta_kubernetes_endpoint_address_target_name"}},
 			{Key: "separator", Value: ";"},
 			{Key: "regex", Value: "Node;(.*)"},
 			{Key: "replacement", Value: "${1}"},
 			{Key: "target_label", Value: "node"},
 		},
-		{
+		{ // Relabel pod labels for pre v2.3 meta labels
```

Comment:
slight change as these labels are only available _starting_ v2.3
```suggestion
		{ // Relabel pod labels for >=v2.3 meta labels
```

Reply:
Ok, I thought it was the other way around, since I did not see them on my discovered targets. I see them on some other targets though. 
Then it might not be certain that this fixes the issue I referenced.